pie/storage/__init__.py

```diff
@@ -0,0 +1,80 @@
+from __future__ import annotations
+
+from nextcord.ext import commands
+
+from pie.storage.database import Storage
+
+
+def get_str(
+    module: commands.Cog, key: str, guild_id: int = 0, def_val: str = None
+) -> str:
+    value = Storage.get(module.qualified_name, key, guild_id)
+    if not value:
+        return def_val
+
+    return value
+
+
+def get_int(
+    module: commands.Cog, key: str, guild_id: int = 0, def_val: int = None
+) -> int:
+    value = Storage.get(module.qualified_name, key, guild_id)
+    if not value:
+        return def_val
+
+    try:
+        value = int(value)
+    except ValueError as e:
+        if def_val:
+            return def_val
+        else:
+            raise e
+
+    return value
+
+
+def get_float(
+    module: commands.Cog, key: str, guild_id: int = 0, def_val: float = None
+) -> float:
+    value = Storage.get(module.qualified_name, key, guild_id)
+    if not value:
+        return def_val
+
+    try:
+        value = float(value)
+    except ValueError as e:
+        if def_val:
+            return def_val
+        else:
+            raise e
+
+    return value
+
+
+def get_bool(
+    module: commands.Cog, key: str, guild_id: int = 0, def_val: bool = None
+) -> bool:
+    value = Storage.get(module.qualified_name, key, guild_id)
+    if not value:
+        return def_val
+
+    if value.lower() not in ["true", "false"]:
+        if def_val:
+            return def_val
+
+        raise ValueError(f"Value {value} for key {key} is not bool!")
+
+    value = value.lower() == "true"
+
+    return value
```

Comment:
I'm not completely against, but wouldn't it be easier if we stored the type in a column and converted back implicitly? It would be much simpler and cleaner to use.

Reply:
Well, we can use some ENUM and store the type, but I think this is better way. I've inspired in Java JSON library which used similar functions to get / store data in JSON. 

Also I think there can be problem in implicit converting since you have to explicitly check every time what you've got for value (or catch ValueError).

Reply:
Java builds on different assumption than Python...

What implicit conversion are you talking about? If we store the data type as column, we can refer to it and convert the value automatically when we get it from the database (wher it is stored as str). So each value will know what its type is.

Reply:
Yes, but then function `get(...)` would return misterious "Object", which is kinda obscure to me.

Reply:
After Discord convo changed the way this works.